[tanshiyu1999] iP

tanshiyu1999 opened the pull request:
# Project Duke
> My name is Bob.

What I can do
- List Tasks
- Add Tasks 
- Edit & Delete Task

Things you need to know (about me)
1. The best Duke Bot
1. My name is Bob
1. Smile More
1. I am ~Pikachu~ Bob

```
running in the 90s
```

- [x] #739
- [ ] https://github.com/octo-org/octo-repo/issues/740
- [ ] Add delight to the experience when all tasks are complete :tada:

This site was built using `Java` with guides from [CS2103T website](https://nus-cs2103-ay2324s1.github.io/).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

tanshiyu1999 committed:
Level-0

tanshiyu1999 committed:
Level-1

tanshiyu1999 committed:
Level-2

tanshiyu1999 committed:
Level-3

tanshiyu1999 committed:
Level-4

tanshiyu1999 committed:
A-TextUiTesting

tanshiyu1999 committed:
Level-5

tanshiyu1999 committed:
Level-6

tanshiyu1999 committed:
A-Enums

tanshiyu1999 committed:
Improvement to Error Handling & Enum

tanshiyu1999 committed:
Level-7 code done

tanshiyu1999 committed:
Merge branch 'branch-Level-7'

Level 7 Completed

tanshiyu1999 committed:
Level-8 Branch Minimal Task Done

tanshiyu1999 committed:
Merge branch 'branch-Level-8'

tanshiyu1999 committed:
A-MoreOOP

tanshiyu1999 committed:
Merge tag 'A-MoreOOP'

tanshiyu1999 committed:
A-Packages

tanshiyu1999 committed:
Merge branch 'add-gradle-support'

tanshiyu1999 committed:
A-Gradle

tanshiyu1999 committed:
A-JUnit

tanshiyu1999 committed:
A-Jar

tanshiyu1999 committed:
A-JavaDoc

tanshiyu1999 committed:
A-CodingStandard

tanshiyu1999 committed:
Level-9

tanshiyu1999 committed:
A-CheckStyle

tanshiyu1999 committed:
GUI setup complete

tanshiyu1999 committed:
Level-10

tanshiyu1999 committed:
Added GUI closure after bye command

leezhanpeng commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,90 @@
+    }
+
+    /**
+     * This method prints a horizonyal line.
```
Just a small typo here.

leezhanpeng commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,90 @@
+
+    }
+
+    /**
+     * Mark the duke.task as undone.
+     */
+    public String markUndone() {
+        String strToPrint = "OK, I've marked this duke.task as not done yet:\n";
+        this.isDone = false;
+        return strToPrint + ("[" + this.getStatusIcon() + "] " + this.getDescription());
+    }
```
Maybe adding a @return tag here is relevant as the return of the string may not come naturally for this function?

leezhanpeng commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,50 @@
+        }
+    }
+
+    /**
+     * Default System.out when this function is called.
+     * @return string information in the Event instance.
+     */
```
For @return tags in Javadoc, it is actually indicated to be optional for members like toString() as they are [self-explanatory](https://google.github.io/styleguide/javaguide.html#s7.3.1-javadoc-exception-self-explanatory), so there is no problem to give one-liners like this:
> /**
> &ensp;\* Returns the string information in the Event instance.
> &ensp;*/

There are some other files with the same case.

leezhanpeng commented on src/main/java/duke/taskList/TaskList.java:
```diff
@@ -0,0 +1,358 @@
+    }
+
+
+    /**
+     * Adds a event into ArrayList.
+     * @param command The duke.command specifying what Event to add.
+     */
```
Again, a @return tag here might be useful?

Other instances of @return tag can be added to other functions in this file as well!

leezhanpeng reviewed (commented):
LGTM. I didn't manage to spot any concerns, only some nitpicks. Great job!

quzhetao01 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,60 @@
+     * Load the data from duke.txt into the TaskList inside of the program.
+     * @param list the TaskList that the data is loaded to from duke.txt.
+     */
+    public void load(TaskList list) {
```
Can afford to be more specific here by specifying what you will be loading (Eg. loadTaskList)

quzhetao01 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,60 @@
+            while (myReader.hasNextLine()) {
+                String data = myReader.nextLine();
+                String type = data.substring(0, 1);
+                switch (type) {
```
The `switch-case` statement needs to follow the coding standard stated in the course (no indentation for the case clause).

quzhetao01 commented on src/main/java/duke/taskList/TaskList.java:
```diff
@@ -0,0 +1,358 @@
+
+
+    /**
+     * Add a Todo into the ArrayList.
```
@return tag is missing here

quzhetao01 commented on src/main/java/duke/taskList/TaskList.java:
```diff
@@ -0,0 +1,358 @@
+                }
+            }
+
+            if (hasBy == false) {
```
abit of nitpicking, but maybe we can shorten this to !hasBy

quzhetao01 reviewed (commented):
Javadoc comments require some improvements. Naming-wise, mostly well done other than some nitpicking. Remember the switch-case coding standard for CS2103T course!

evanyan13 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,69 @@
+        return "Duke heard: " + input;
+    }
+
+    public String getCommandResponse(String command) {
```
Maybe can include JavaDoc for this method to differentiate from the `getResponse()` method above?

evanyan13 commented on src/main/java/duke/taskList/TaskList.java:
```diff
@@ -0,0 +1,358 @@
+        }
+
+        try {
+            FileWriter myWriter = new FileWriter("./src/main/data/duke.txt", true);
```
Is it possible to store the address of the input/output file as a constant somewhere instead of hardcoding them every time? It might be easier for you to edit it as well if the file location changes afterwards

```suggestion
            FileWriter myWriter = new FileWriter(FILE_PATH, true);
```

evanyan13 reviewed (commented):
LGTM! Only identified a few nits in my comments 👍

tanshiyu1999 committed:
Assertion added

tanshiyu1999 committed:
Merge pull request #1 from tanshiyu1999/branch-A-Assertions

Assertion added

tanshiyu1999 committed:
Quality code

tanshiyu1999 committed:
Merge pull request #2 from tanshiyu1999/branch-A-CodeQuality

Quality code

tanshiyu1999 committed:
Reminder

tanshiyu1999 committed:
add tag

tanshiyu1999 committed:
Merge pull request #3 from tanshiyu1999/branch-B-Reminders

Branch b reminders

tanshiyu1999 committed:
Added a new feature to direct people to a README

Added a new command "help" to direct users to my github page, where the README.md file is located

The README file will contain all of the instructions and guidance required for users to know what command they can use when utilizing Bob.

The "help" command is chosen as it is something intuitive for users to type in when they are unfamiliar with the command required.

tanshiyu1999 committed:
Merge pull request #5 from tanshiyu1999/branch-B-Reminders

Added a new feature to direct people to a README

tanshiyu1999 committed:
Assert tag

tanshiyu1999 committed:
clear tag bug

tanshiyu1999 committed:
w

tanshiyu1999 committed:
This is to attempt to update my CodeQuality

A secondary check has been done to my code quality, to ensure that most of the segments are following the code quality guidelines as specified by the
2013T course.

This also seeks to add in the code quality tag, which I forgot to add in in the past.

tanshiyu1999 committed:
A-UserGuide

Created a User Guide for new users to refer to when using and interacting with Bob!

tanshiyu1999 committed:
Added Ui.png

tanshiyu1999 committed:
w

tanshiyu1999 committed:
Updated README.md

tanshiyu1999 committed:
Add README.me in docs

tanshiyu1999 committed:
A-Release

tanshiyu1999 committed:
w

tanshiyu1999 committed:
fixing readme

tanshiyu1999 committed:
Database stores marked

tanshiyu1999 committed:
Error msg bug fix

tanshiyu1999 committed:
New Release
